src/pages/MainPage.tsx

```diff
@@ -12,16 +12,21 @@ import styled from '@emotion/styled';
 import { theme } from '@/styles/theme';
 import { useThemes } from '@/hooks/useThemes';
 import { useRankingProducts } from '@/hooks/useRankingProducts';
+import { useNavigate } from 'react-router-dom';
 
 const DEFAULT_TARGET = 'ALL';
 const DEFAULT_RANK = 'MANY_WISH';
 
 function ThemesSection() {
   const { themes, loading, error } = useThemes();
+  const navigate = useNavigate();
   if (loading) return <Spinner />;
   if (error) return null;
   if (themes.length === 0) return null;
-  return <GiftThemeGrid themes={themes} />;
+  const handleThemeClick = (theme: import('@/types').GiftTheme) => {
+    navigate(`/themes/${theme.themeId}`);
```

Comment:
[참고] 경로를 템플릿 리터럴로 리터럴로 생성하면 휴먼에러가 발생할 가능성이 있어요.

React Router의 [generatePath](https://reactrouter.com/6.30.1/utils/generate-path) 유틸을 활용하면 경로 패턴을 상수로 관리하면서 매개변수를 안전하게 주입할 수 있어 편하더라구요.

```suggestion
    const THEME_PATH = '/themes/:themeId';

    navigate(generatePath(THEME_PATH), {
        themesId: theme.themeId
    });
```